flanker/addresslib/address.py

```diff
@@ -99,14 +100,9 @@ def parse(address, addr_spec_only=False, metrics=False):
         bstart = time()
         retval = _lift_parser_result(parser.parse(address, lexer=lexer.clone()))
         mtimes['parsing'] = time() - bstart
-    except LexError as e:
-        log.warning(u'error in lexing: %s', e)
-        return None, mtimes
-    except YaccError as e:
-        log.warning(u'error in parsing: %s', e)
-        return None, mtimes
-    except SyntaxError as e:
-        log.warning(u'error in parsing: %s', e)
+    except:
+        log.warning('Failed to parse address: %s',
+                    address.decode('utf-8', 'replace'))
```

Comment:
Looks good but we should also fix this in `parse_discrete_list` [here](https://github.com/mailgun/flanker/blob/master/flanker/addresslib/address.py#L155).